fabric-gamerules-v1/src/main/java/net/fabricmc/fabric/impl/gamerule/GameRuleUtilsImpl.java

```diff
@@ -35,10 +37,13 @@
 	@Override
 	public <T extends GameRules.Rule<T>> GameRules.RuleType<T> createCustomRule(Supplier<ArgumentType<?>> argumentType, Function<GameRules.RuleType<T>, T> factory, BiConsumer<MinecraftServer, T> notifier) {
 		try {
-			java.lang.reflect.Constructor<GameRules.RuleType> constructor = GameRules.RuleType.class.getDeclaredConstructor(Supplier.class, Function.class, BiConsumer.class);
-			constructor.setAccessible(true);
-			return constructor.newInstance(argumentType, factory, notifier);
-		} catch (IllegalAccessException | InstantiationException | ClassCastException | NoSuchMethodException | InvocationTargetException e) {
+			if (typeConstructor == null) {
+				java.lang.reflect.Constructor<GameRules.RuleType> constructor = GameRules.RuleType.class.getDeclaredConstructor(Supplier.class, Function.class, BiConsumer.class);
```

Comment:
Why qualified import here

Reply:
Huh. I thought I'd gotten rid of that earlier; Idea was throwing a minor fit over imports while I was writing so I qualified it but it's fine now so I'll change it back.